### PR TITLE
feat: serialize RouteList for Trips

### PR DESCRIPTION
**Pull request by lucys-s**
Modify SerializedTrip and TripSerializer to add RouteList field that takes in a list of SerializedLatLng. This data will allow the sample app to display polyline with traffic condition.

### CODE REVIEW

File: src/main/java/com/example/provider/json/SerializedTrip.java [hunk 1]
```diff
@@ -33,6 +34,8 @@ abstract class SerializedTrip {
   abstract ImmutableList<Waypoint> waypoints();
   /** ID of the vehicle. */
   abstract String vehicleId();
+  /** Vehicle route list. */
+  abstract List<LatLng> routeList();
```
Comment (ShirwaM) [inline src/main/java/com/example/provider/json/SerializedTrip.java:38]: nit: this is actually for the trip so can we use "Trip route list."?

Comment (ShirwaM) [inline src/main/java/com/example/provider/json/SerializedTrip.java:38]: Let's use `SerializedLatLng` here so the client can use the fields "latitude" and "longitude" without the "_" suffix.

**Review by ShirwaM — CHANGES_REQUESTED** [review]

**Comment by ArturoSalazarB16** [comment]
Nit: lets add a bit more context to the PR title, example: 'feat: serialize RouteList for Trips/Vehicles'

And also let's remove the part that mentions 'CurrentRouteSegmentTraffic' from the description.

**Review by ArturoSalazarB16 — APPROVED** [review]

**Comment by ArturoSalazarB16** [comment]
Nit: just confirming, did we test that existing single destination trip scenario works with this change?

File: src/main/java/com/example/provider/json/TripSerializer.java [hunk 2]
```diff
@@ -69,9 +71,20 @@ public JsonElement serialize(Trip src, Type typeOfSrc, JsonSerializationContext
             .add(dropoffWaypoint)
             .build();
 
+    List<SerializedLatLng> serializedRouteList = new ArrayList<SerializedLatLng>();
```
Comment (ArturoSalazarB16) [inline src/main/java/com/example/provider/json/TripSerializer.java:74]: Nit: consider a 'Functional programming' approach here by mapping the list (We do something similar right above with 'intermediateDestinations'

```java
List<SerializedLatLng> serializedRouteList = src.getRouteList().stream().map(route -> 
  SerializedLatLng.newBuilder()
                .setLatitude(route.getLatitude())
                .setLongitude(route.getLongitude())
                .build()).collect(Collectors.toList());
```

**Comment by lucys-s** [comment]
> Nit: just confirming, did we test that existing single destination trip scenario works with this change?

Yes. In trip response, "routeList" will be added, and the corresponding value is a list SerializedLatLng routes with fields "latitude" and "longitude". The route displays correctly with this change

**Review by ShirwaM — APPROVED** [review]

**Pull request merged**